pkg/reconciler/knativeserving/knativeserving.go

```diff
@@ -117,21 +119,32 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ks *servingv1alpha1.Knat
 func (r *Reconciler) transform(ctx context.Context, instance *servingv1alpha1.KnativeServing) (mf.Manifest, error) {
 	logger := logging.FromContext(ctx)
 
-	logger.Debug("Transforming manifest")
+	logger.Debug("Adding ingress resources")
+	ingressManifest, err := ingress.Manifest(ctx, instance)
+	if err != nil {
+		return mf.Manifest{}, err
+	}
 
+	// TODO: This is is a little backwards. We might wanna fix this in manifestival.
+	combined, err := mf.ManifestFrom(mf.Slice(append(r.config.Resources(), ingressManifest.Resources()...)),
+		mf.UseClient(r.config.Client))
+	if err != nil {
+		return mf.Manifest{}, err
+	}
```

Comment:
@trshafer here's a good argument for adding an `Add` function to the interface, if only to keep from having to pass the client and logger options along. https://github.com/manifestival/manifestival/issues/41

Reply:
I was using the readme example yesterday and it took me a while to realize that I needed to pass the client and there was no way to pass the logger. It was purely for debugging as one of the unstructured wasn't properly transforming, so I wanted to pull the resources out and apply them one by one with detailed debugging. I could see `manifest.NewManifest(Slice)`.

Reply:
@jcrossley3 I restructured the code a bit. Arguably it makes more sense now, as the mfClient isn't only used in the manifest's context today (it's also used for a transformer).

As we're in parallel moving towards decoupling operator and operand version, the reconcile loop will have to do the loading of the core manifest anyway. Moving it there now kinda solves the awkwardness of having to carry over client and logger as we're just building it there. Please take another look.